src/tools/compiletest/src/header.rs

```diff
@@ -286,6 +286,9 @@ pub struct TestProps {
     // directory as the test, but for backwards compatibility reasons
     // we also check the auxiliary directory)
     pub aux_builds: Vec<String>,
+    // A list of crates to pass '--extern-private name:PATH' flags for
+    // This should be a subset of 'aux_build'
+    pub extern_private: Vec<String>,
```

Comment:
https://github.com/rust-lang/rust/pull/59335#issuecomment-475857324

Reply:
I believe that it makes sense to include some actual testing of `--extern-private` in this PR (especially considering that my initial PR had uncaught issues, necessitating this PR). In a future PR, this could be removed in favor of a more robust solution.

Reply:
The option parsing part can be tested on `=/path/to/nowhere` and other testing doesn't need the `=PATH` component after `--extern-private name`.

Reply:
The reason I made this PR is that `--extern-private` with a path currently doesn't work at all. I think it would be a really bad idea for this PR to contain no testing of the core feature that it adds (`--extern-private` with a *valid* path)

Reply:
OK, could you add a big FIXME to this then, linking to https://github.com/rust-lang/rust/pull/54020?

Reply:
Added